collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -92,7 +93,36 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     private void initToolbar() {
         mToolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(mToolbar);
+        mToolbar.inflateMenu(R.menu.menu);
+
+        mToolbar.findViewById(R.id.menu_sort).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mDrawerLayout.isDrawerOpen(Gravity.END)) {
+                    mDrawerLayout.closeDrawer(Gravity.END);
+                } else {
+                    Collect.getInstance().hideKeyboard(mInputSearch);
+                    mDrawerLayout.openDrawer(Gravity.END);
+                }
+            }
+        });
+
+        mToolbar.findViewById(R.id.menu_filter).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSearchBoxLayout.getVisibility() == View.GONE) {
+                    showSearchBox();
+                } else {
+                    hideSearchBox();
+                }
+            }
+        });
+
+        boolean hasHardwareMenu =
```

Comment:
You are absolutely correct. But this UI is gonna be changed soon. So, this will be replaced with a better approach soon along with the newer UI.